runtime/lua/vim/lsp/buf.lua

```diff
@@ -143,6 +143,48 @@ local function select_client(method, on_choice)
   end
 end
 
+--- Formats the current buffer.
+---@param opts table
+---     formatting_options: table: Can be used to specify FormattingOptions.
+---                                Some unspecified options will be automatically derived from the current
+---                                Neovim options.
+---                                @see https://microsoft.github.io/language-server-protocol/specification#textDocument_formatting
+---     timeout_ms: string
+---     filter: function
+---     ids: table
+---     names: table
+---     id: string
+---     name: string
+
+function M.format(opts)
+  local bufnr = vim.api.nvim_get_current_buf()
```

Comment:
I see the other formatting functions don't accept a bufnr, but what about considering an optional `opts.bufnr` here?

Reply:
I think this is a footgun and I don't really see the usecase. I added it for now but am probably going to remove it.

Reply:
Yeah it does feel a bit "forced". Reason I suggested it is that I've personally ran into multiple cases (with certain plugins) where it would've been nice to be able to provide a bufnr other than the current one. I could picture some use cases for programmatically formatting multiple buffers at once, based on some condition/trigger, like:

```lua
for _, bufnr in ipairs(vim.api.nvim_list_bufs()) do
    local filetype = vim.api.nvim_buf_get_option(bufnr, "filetype")
    if filetype == "typescript" then
        vim.lsp.buf_format(bufnr, { name = "eslint" })
    end
end
```

I also just realized this method is under the `vim.lsp.buf.` table, for which it'd make sense not to accept a bufnr. If it were to remain a parameter, maybe it'd make sense to move the function to `vim.lsp.buf_format` with a slightly changed signature (e.g. `vim.lsp.buf_format(bufnr, { name = "eslint" })`).

Reply:
I don't like passing mixed signatures as it prevents calling the function directly with a table